common/tools/dev-tool/src/commands/samples/checkNodeVers.ts

```diff
@@ -0,0 +1,323 @@
+\tcp ${envFilePath} ${containerWorkspacePath}/samples/javascript/;
+}
+
+function run_samples() {
+\tsamples_path=\$1;
+\techo "Using node \$(node -v) to run samples in \${samples_path}";
+\tcd "\${samples_path}";
+\tfor SAMPLE in *.js; do
+\t\tnode \${SAMPLE};
+\tdone
+}
+
+function build_typescript() {
+\techo "Using node \$(node -v) to build the typescript samples";
+\tcd ${containerWorkspacePath}/samples/typescript
+\t${compileCMD(`npm run build`, printToScreen)}
+\tcp ${envFilePath} ${containerWorkspacePath}/samples/typescript/dist/
+}
+
+function main() {
+\tinstall_packages;
+\trun_samples "${javascriptSamplesPath}";
+\tbuild_typescript && run_samples "${typescriptCompiledSamplesPath}";
+}
+
+main`;
+  return scriptContent;
+}
+
+function createDockerContextDirectory(
+  dockerContextDirectory: string,
+  containerWorkspacePath: string,
+  samples_path: string,
+  envPath: string,
+  packagePath?: string,
+  logFilePath?: string,
+  callback?: () => void
+): void {
+  if (packagePath === undefined) {
+    throw new Error("package_path is a required argument but it was not passed");
+  }
+  const packageName = path.basename(packagePath);
+  const envFileName = path.basename(envPath);
+  fs.copySync(samples_path, path.join(dockerContextDirectory, "samples"));
+  fs.copyFileSync(packagePath, path.join(dockerContextDirectory, packageName));
+  fs.copyFileSync(envPath, path.join(dockerContextDirectory, envFileName));
+  fs.writeFileSync(
+    path.join(dockerContextDirectory, "run_samples.sh"),
+    buildRunSamplesScript(containerWorkspacePath, packageName, envFileName, logFilePath),
+    { mode: S_IRWXO }
+  );
+  callback?.();
+}
+
+function runDockerContainer(
+  dockerContextDirectory: string,
+  dockerImageName: string,
+  dockerContainerName: string,
+  containerWorkspace: string,
+  stdoutListener: (chunk: string | Buffer) => void,
+  stderrListener: (chunk: string | Buffer) => void,
+  callback?: () => void
+) {
+  const args = [
+    "run",
+    "--name",
+    dockerContainerName,
+    "--workdir",
+    containerWorkspace,
+    "-v",
+    `${dockerContextDirectory}:${containerWorkspace}`,
+    dockerImageName,
+    "./run_samples.sh"
+  ];
+  const dockerContainerRunProcess = pr.spawn("docker", args, {
+    cwd: dockerContextDirectory
+  });
+  log.info(`Started running the docker container ${dockerContainerName}`);
+  dockerContainerRunProcess.stdout.on("data", stdoutListener);
+  dockerContainerRunProcess.stderr.on("data", stderrListener);
+  dockerContainerRunProcess.on("exit", function(code) {
+    if (code === 0) {
+      log.info("Docker container finished running");
+      callback?.();
+    } else {
+      log.error("Docker container encountered an error");
+    }
+  });
+}
+
+export const commandInfo = makeCommandInfo(
+  "checkNodesVers",
+  "execute samples with different node versions, typically in preparation for release",
+  {
+    package_path: {
```

Comment:
These should be kebab-cased by convention. E.g.

`dev-tool samples check-release --node-versions "8,10,12,14"`

Reply:
Man you made me crave it this late in night ...